lib/nostrum/struct/snowflake.ex

```diff
@@ -111,12 +158,13 @@ defmodule Nostrum.Struct.Snowflake do
   #DateTime<2016-05-05 21:04:13.203Z>
   ```
   """
-  @spec creation_time(t) :: DateTime.t()
+  @spec creation_time(t) :: DateTime.t() | no_return
   def creation_time(snowflake) when is_snowflake(snowflake) do
     use Bitwise
 
-    time_elapsed_ms = (snowflake >>> 22) + 1_420_070_400_000
+    time_elapsed_ms = (snowflake >>> 22) + Constants.discord_epoch()
 
-    DateTime.from_unix!(time_elapsed_ms, :milliseconds)
+    {:ok, datetime} = DateTime.from_unix(time_elapsed_ms, :milliseconds)
```

Comment:
By changing this from a banged function, we no longer need the `no_return`. I think the function was fine before, it just needed the `no_return`.

Reply:
I've removed the `no_return` in a6375f3.

Reply:
As per our talk in the discord channel, I've ran the math on the function. It can't fail. However, I'm also adding in some tests to make sure it stays that way. This ensures that f we ever do change the implementation, we'll know if the function can break.

See 0fd0cc8.